### PR TITLE
Add support for GiEX QT06 valve

### PR DESCRIPTION
**Pull request by mbo18**
Add support for GiEX QT06 valve

Fixes: https://github.com/zigpy/zha-device-handlers/issues/1622

### CODE REVIEW

**Comment by MattWestb** [comment]
Use https://black.vercel.app/ copy and pace in the right and you is getting the fixed code in the left side.

**Comment by MattWestb** [comment]
My record with black is over 10 time fixing for getting all right  ;-))

**Comment by mbo18** [comment]
> My record with black is over 10 time fixing for getting all right ;-))

Would have been 10+ without your link, thanks 👍

**Review by javicalle — CHANGES_REQUESTED** [review]
Just some gramatical changes to remove declaration overload.

File: zhaquirks/tuya/ts0601_valve.py [hunk 1]
```diff
@@ -251,3 +257,153 @@ class ParksidePSBZS(EnchantedDevice):
...
+GIEX_TUYA_VALVE_START_TIME_ATTR = 0xEF65  # Last irrigation start time (GMT)
+GIEX_TUYA_VALVE_END_TIME_ATTR = 0xEF66  # Last irrigation end time (GMT)
+GIEX_TUYA_VALVE_NUM_TIMES_ATTR = (
+    0xEF67  # Number of cycle irrigation times, 0 for single cycle, min=0 max=100
+)
+GIEX_TUYA_VALVE_TARGET_ATTR = 0xEF68  # Irrigation target, duration in seconds or capacity in litres (depending on mode), min=0, max=3600
+GIEX_TUYA_VALVE_INTERVAL_ATTR = (
+    0xEF69  # Cycle irrigation interval in seconds, min=0, max=3600
+)
+GIEX_TUYA_VALVE_DURATION_ATTR = 0xEF72  # Last irrigation duration
```
[the remark this answers is not in the document]
Comment (mbo18) [reply zhaquirks/tuya/ts0601_valve.py]: You are right, change done

File: zhaquirks/tuya/ts0601_valve.py [hunk 2]
```diff
@@ -251,3 +257,153 @@ class ParksidePSBZS(EnchantedDevice):
...
+
+
+class GiexTuyaValveManufCluster(TuyaMCUCluster):
+    """GiEX Tuya valve manufacturer cluster."""
```
[the remark this answers is not in the document]
Comment (mbo18) [reply zhaquirks/tuya/ts0601_valve.py]: Done

File: zhaquirks/tuya/ts0601_valve.py [hunk 3]
```diff
@@ -251,3 +257,153 @@ class ParksidePSBZS(EnchantedDevice):
...
+    async def write_attributes(self, attributes, manufacturer=None):
+        """Overwrite to force manufacturer code."""
+
+        return await super().write_attributes(
+            attributes, manufacturer=foundation.ZCLHeader.NO_MANUFACTURER_ID
+        )
+
+
+class GiexTuyaValve(CustomDevice):
+    """GiEX Tuya valve device."""
```
[the remark this answers is not in the document]
Comment (mbo18) [reply zhaquirks/tuya/ts0601_valve.py]: Done

**Review by javicalle — APPROVED** [review]
LGTM

**Comment by javicalle** [comment]
BTW, Z2M is also listening (and dimissing) DP106. Do you have reports from it in your device?
`giexWaterValveCurrentTempurature: 106,`

**Comment by mbo18** [comment]
I did not test it as Z2M is dismissing it. Will do next week and open a new PR if needed

**Comment by javicalle** [comment]
Sure, no problem. Just curious about what values is reporting (if none).

**Comment by mbo18** [comment]
> Sure, no problem. Just curious about what values is reporting (if none).

All I have is "None", so no value

**Comment by Copyrighter** [comment]
any update guys? someone can complete support for this product?

https://zigbee.blakadder.com/GiEX_QT06.html
https://www.zigbee2mqtt.io/devices/QT06_1.html

**Pull request merged**